cloud/migrate-mst-cloud/schema-then-data.md

```diff
@@ -0,0 +1,367 @@
+
+Migrate your pre-data from Managed Service for TimescaleDB to Timescale Cloud.
+This includes table and schema definitions, as well as information on sequences,
+owners, and settings. This doesn't include Timescale-specific schemas. 
+
+### Migrating schema pre-data
+
+<procedure>
+
+1. Dump schema pre-data from Managed Service for TimescaleDB into a .bak file.
+   Exclude Timescale-specific schemas:
+
+   ```bash
+   # Use your Managed Service for TimescaleDB connection details in place of 
+   # <mst_host> and <mst_port>.
+   pg_dump -U tsdbadmin -W -h <mst_host> -p <mst_port> -Fc -v \
+   --section=pre-data --exclude-schema="_timescaledb*" \
+   -f <file-name-pre-data>.bak defaultdb
+   ```
+
+   The service prompts you for a password. Use your credentials for Managed
+   Service in TimescaleDB.
+
+1. Restore the dumped schema pre-data into your Cloud database. Use your Cloud
+   credentials, and set the `--no-owner` flag to avoid permissions problems:
+
+    ```bash
+    # Use your Cloud credentials and connection details in place of <cloud_host> 
+    # and <cloud_port>.
+    pg_restore -U tsdbadmin -W -h <cloud_host> -p <cloud_port> --no-owner -Fc \
+    -v -d tsdb <file-name-pre-data>.bak
+    ```
+
+</procedure>
+
+<highlight type="note">
+You may encounter the following errors while running `pg_restore`. You can
+safely ignore them. The migration still occurs successfully.
+</highlight>
+
+`pg_restore` throws a permissions error when it tries to apply the TimescaleDB
+extension. Because TimescaleDB is already installed by default on Timescale
+Cloud, you can safely ignore this:
+
+```bash
+pg_restore: creating EXTENSION "timescaledb"
+pg_restore: creating COMMENT "EXTENSION timescaledb"
+pg_restore: while PROCESSING TOC:
+pg_restore: from TOC entry 6239; 0 0 COMMENT EXTENSION timescaledb 
+pg_restore: error: could not execute query: ERROR:  must be owner of extension timescaledb
+```
+
+If you have continuous aggregates, you may get the following error. Ignore this
+as you will [restore your aggregates](#recreate-continuous-aggregates) in a
+later procedure.
+
+```bash
+pg_restore: error: could not execute query: ERROR:  relation "_timescaledb_internal._materialized_hypertable_x" does not exist
+```
+
+You can also ignore:
+
+```bash
+​​pg_restore: WARNING:  no privileges were granted for "<..>"
+```
+
+## Restore hypertables in Timescale Cloud
+
+After pre-data migration, your hypertables from Managed Service for TimescaleDB
+become regular PostgreSQL tables in Cloud. Recreate your hypertables in Cloud to
+restore them.
+
+### Restoring hypertables in Timescale Cloud
+
+<procedure>
+
+1. Connect to your Cloud database using psql:
+
+   ```bash
+   # Use your Cloud credentials and connection details.
+   psql “postgres://tsdbadmin:<cloud_password>@<cloud_host>:<cloud_port>/tsdb?sslmode=require”
+   ```
+
+1. In psql, run:
+
+   ```bash
+   SELECT create_hypertable(
+       '<table-name>', '<time-column-name>', 
```

Comment:
```suggestion
       '<table-name>', '<time-column-name>', 
```
I'd suggest staying consistent with variable naming across these docs.

e.g. `variable_name` vs. `variable-name`

Reply:
sorry, I know this is how I documented it in my initial outline doc

Reply:
```suggestion
       '<table_name>', '<time_column_name>', 
```

I think this is what you meant right? 😄